### ISSUE TITLE
Axios 0.15.2 doesn't reject promise if request is cancelled in Chrome

### ISSUE THREAD

**Issue by michaeluskov** [axios/axios#537]
I have axios@0.15.2

If this HTML is run, some requests will be cancelled by Chrome:

    <html>
    <head>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/axios/0.15.2/axios.js"></script>
    </head>
        <script>
        var promises = [];
        window.onload = function() {
            for (var i = 0; i < 4; i++) {
                var promise = axios.get('some url which causes ERR_CONNECTION_CLOSED');
                promises.push(promise);
            }
        }
       </script>
    </html>

![Cancelled](https://cloud.githubusercontent.com/assets/4506837/20349972/9123cef0-ac2e-11e6-93db-0ec66f205382.png)

But promises which turned into cancelled request won't be fulfilled:

![notfulfilled](https://cloud.githubusercontent.com/assets/4506837/20350023/d2146c4e-ac2e-11e6-89bc-983233628d09.png)

**Comment by mzabriskie** [comment]
Can you share the code that is generating the cancelled requests?

**Comment by fvicente** [comment]
Hi @mzabriskie 
I have exactly the same problem and I can reproduce it consistently. My app has a long pull ajax, if I put my laptop to sleep, when it comes back the ajax es cancelled by chrome but never completed/rejected the promise.

code is like this:
```javascript
  const timeStamp = (new Date()).getTime();
  return axios.get(`${API_LISTEN_URL}?subject=POS${posId}&syncId=${syncId}&_ts=${timeStamp}`, {
    headers: defaultHeaders,
    responseType: 'document'
  })
  .then(response => {
    // we need to retrieve the new synchronization ID - this needs to be sent on the next request
    return { 'ok': true, 'syncId': response.headers["x-sync-id"], 'xml': response.data }
  }, error => {
    const message = "Error while listening for an event"
    console.log(message)
    return { 'ok': false }
  })
```

You have to put some sleep on your server side in order to reproduce the problem. Then when the ajax is pending, close your laptop monitor, after a few seconds lift and you will see that the request is cancelled, but the promise not resolved.
Note: I'm using chrome Version 55.0.2883.95 (64-bit) on a MacBook computer, MacOS X "El Capitán" version 10.11.6 (15G1217)

**Comment by fvicente** [comment]
Pull request created to fix this issue: https://github.com/mzabriskie/axios/pull/689

**Comment by mattridley** [comment]
I'm encountering the same issue - given @fvicente is no longer working on a PR to resolve this does anyone have an objection if I pick the issue up and run with it?

**Comment by fvicente** [comment]
Hey @mattridley 
I'm not saying is the best solution, but you can always monkey-patch the library, by putting in your code something like this:

```javascript
  try {
    // monkey patch for "onAbort" not being handled by axios correctly, this is to avoid an
    // unresolved promise on the long-pull ajax if the computer goes to sleep
    // https://github.com/axios/axios/pull/689
    const oldSend = XMLHttpRequest.prototype.send
    XMLHttpRequest.prototype.send = function () {
      this.onabort = this.onerror
      oldSend.apply(this, arguments)
    }
  } catch (e) {
    console.error(e)
  }
```

**Comment by nickuraltsev** [comment]
Fixed via #1399
